src/System.Text.Json/src/System/Text/Json/JsonHelpers.Date.cs

```diff
@@ -48,33 +67,29 @@ public static bool TryParseAsISO(ReadOnlySpan<byte> source, out DateTime value)
         /// <returns>"true" if successfully parsed.</returns>
         public static bool TryParseAsISO(ReadOnlySpan<byte> source, out DateTimeOffset value)
         {
-            return TryParseDateTimeOffset(source, out value, out _);
-        }
+            DateTimeParseData parseData = new DateTimeParseData();
 
-        private struct DateTimeParseData
-        {
-            public int Year;
-            public int Month;
-            public int Day;
-            public int Hour;
-            public int Minute;
-            public int Second;
-            public int Fraction; // This value should never be greater than 9_999_999.
-            public int OffsetHours;
-            public int OffsetMinutes;
-            public bool OffsetNegative => OffsetToken == JsonConstants.Hyphen;
-            public byte OffsetToken;
+            if (!TryParseDateTimeOffset(source, ref parseData))
```

Comment:
Why is `DateTimeParseData` being passed by ref here rather than the `TryParseDateTimeOffset` new-ing one up and returning as an out?